go/src/koding/klientctl/auth/register.go

```diff
@@ -0,0 +1,326 @@
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Register failed with error:", err)
+		log.Error("%s", err)
+		return 1
+	}
+
+	clientID, err := doLoginRequest(client, host, token)
+	if err != nil {
+		// we don't need to inform user about the error after user registered successfully
+		log.Error("%s", err)
+		return 1
+	}
+
+	// Set clientId into the kd.bolt
+	if err = kloud.Cache().Set("clientId", clientID); err != nil {
+		log.Error("error while caching clientId")
+		return 1
+	}
+
+	// team cannot be empty (because of required while registering)
+	// otherwise it return error while registering user
+	// store groupName or slug as "team" inside the cache
+	if err = kloud.Cache().Set("team", c.String("team")); err != nil {
+		log.Error("error while caching team")
+		return 1
+	}
+
+	fmt.Println("This is your session ID, keep it safe.", clientID)
+	return 0
+}
+
+func initRegisterRequest(c *cli.Context) *RegisterRequest {
+	return &RegisterRequest{
+		Username:        c.String("username"),
+		FirstName:       c.String("firstName"),
+		LastName:        c.String("lastName"),
+		Password:        c.String("password"),
+		PasswordConfirm: c.String("passwordConfirm"),
+		Email:           c.String("email"),
+		Slug:            c.String("team"),
+		// if company flag exists, value will be company one
+		// otherwise value would be team's value
+		CompanyName:   selectExistingOne(c, "company", "team"),
+		Newsletter:    c.String("newsletter"),
+		AlreadyMember: c.String("alreadyMember"),
+		Agree:         "on",
+	}
+}
+
+// selectExistingOne selects the argument that existing value
+// first existing value will be chosen all the time
+func selectExistingOne(c *cli.Context, args ...string) string {
+	for _, arg := range args {
+		if c.String(arg) != "" {
+			return c.String(arg)
+		}
+	}
+
+	return ""
+}
+
+func checkAndAskRequiredFields(r *RegisterRequest) (*RegisterRequest, error) {
+	var err error
+
+	if r.Username == "" {
+		r.Username, err = helper.Ask("Username : ")
+		if err != nil {
+			return nil, err
+		}
+	}
+	if r.FirstName == "" {
+		r.FirstName, err = helper.Ask("FirstName : ")
+		if err != nil {
+			return nil, err
+		}
+	}
+	if r.LastName == "" {
+		r.LastName, err = helper.Ask("LastName : ")
+		if err != nil {
+			return nil, err
+		}
+	}
+	if r.Password == "" {
+		r.Password, err = helper.AskSecret("Password [***]: ")
+		if err != nil {
+			return nil, err
+		}
+	}
+	if r.PasswordConfirm == "" {
+		r.PasswordConfirm, err = helper.AskSecret("Confirm Password [***]: ")
+		if err != nil {
+			return nil, err
+		}
+		if r.PasswordConfirm != r.Password {
+			return nil, fmt.Errorf("Different Password/ConfirmPassword ")
+		}
+	}
+	if r.Email == "" {
+		r.Email, err = helper.Ask("Email []: ")
+		if err != nil {
+			return nil, err
+		}
+	}
+	if r.Slug == "" {
+		r.Slug, err = helper.Ask("Team Name []: ")
+		if err != nil {
+			return nil, err
+		}
+	}
+	if r.CompanyName == "" {
+		r.CompanyName, err = helper.Ask("Company Name []: ")
```

Comment:
user might not take companyname's slug if its taken. it should be unique. its also optional case. if user doesn't enter company name it would be slug name as default.